In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.tree import DecisionTreeClassifier, plot_tree
from sklearn.metrics import classification_report, confusion_matrix, accuracy_score,\
f1_score, precision_recall_fscore_support
import prepare

In [2]:
train, validate, test = prepare.wrangle_data('titanic')

In [3]:
train.head()

survived  pclass     sex        age  sibsp  parch   fare  embark_town  \
428         0       3    male  29.699118      0      0   7.75   Queenstown   
480         0       3    male   9.000000      5      2  46.90  Southampton   
888         0       3  female  29.699118      1      2  23.45  Southampton   
822         0       1    male  38.000000      0      0   0.00  Southampton   
612         1       3  female  29.699118      1      0  15.50   Queenstown   

     alone  sex_male  embark_town_Queenstown  embark_town_Southampton  
428      1         1                       1                        0  
480      0         1                       0                        1  
888      0         0                       0                        1  
822      1         1                       0                        1  
612      0         0                       1                        0

## 1. 
What is your baseline prediction? What is your baseline accuracy? remember: your baseline prediction for a classification problem is predicting the most prevelant class in the training dataset (the mode). When you make those predictions, what is your accuracy? This is your baseline accuracy.

In [4]:
train.survived.value_counts()

0    307
1    191
Name: survived, dtype: int64

#### What is your baseline prediction?

survived = 0

In [5]:
# train['baseline'] = 0
# validate['baseline'] = 0
# test['baseline'] = 0

In [6]:
# # What is your baseline accuracy?
# print(f'baseline prediction accuracy score is: \
# {accuracy_score(train.survived, train.baseline):.2%}')

In [7]:
# finding baseline accuracy
baseline_accuracy = (train.survived == 0).mean()

In [8]:
# What is your baseline accuracy?
print(f'baseline prediction accuracy score is: \
{baseline_accuracy:.2%}')

baseline prediction accuracy score is: 61.65%


## 2.
Fit the decision tree classifier to your training sample and transform (i.e. make predictions on the training sample)

In [9]:
X_cols = train.drop(columns=['sex','embark_town','survived']).columns.to_list()
y_cols = 'survived'

In [10]:
X_train = train[X_cols]
y_train = train[y_cols]

In [11]:
# make the decision tree classifier object
clf1 = DecisionTreeClassifier()
# fit the DTC object
clf1.fit(X_train, y_train)
# use the DTC object
model_1_preds = clf1.predict(X_train)

## 3. 
Evaluate your in-sample results using the model score, confusion matrix, and classification report.



In [12]:
# model 1 score
print(f'model 1 accuracy is: {clf1.score(X_train, y_train):.2%}')

model 1 accuracy is: 99.20%


In [13]:
# confusion matrix for model 1
print(f'confusion matrix for model 1: \n{confusion_matrix(y_train, model_1_preds)}')

confusion matrix for model 1: 
[[307   0]
 [  4 187]]


In [45]:
rubric = pd.DataFrame(
    {'pred_0': ['True Neg', 'False Neg'],
     'pred_1': ['False Pos', 'True Pos']
    }, index=['actual_0', 'actual_1'])

In [46]:
conf = confusion_matrix(y_train, model_1_preds)

In [47]:
rubric + ': ' + conf.astype(str)

pred_0         pred_1
actual_0  True Neg: 307   False Pos: 0
actual_1   False Neg: 4  True Pos: 187

In [14]:
# classification report for model 1
print(f'classification report for model 1:\n\n \
{classification_report(y_train, model_1_preds)}')

classification report for model 1:

               precision    recall  f1-score   support

           0       0.99      1.00      0.99       307
           1       1.00      0.98      0.99       191

    accuracy                           0.99       498
   macro avg       0.99      0.99      0.99       498
weighted avg       0.99      0.99      0.99       498



## 4.
Compute: Accuracy, true positive rate, false positive rate, true negative rate, false negative rate, precision, recall, f1-score, and support.

In [15]:
# Accuracy
accuracy_model_1 = accuracy_score(y_train, model_1_preds)

# true positive rate
tp_model_1 = ((y_train == 1) & (model_1_preds == 1)).sum()

# false positive rate
fp_model_1 = ((y_train == 0) & (model_1_preds == 1)).sum()

# true negative rate
tn_model_1 = ((y_train == 0) & (model_1_preds == 0)).sum()

# false negative rate
fn_model_1 = ((y_train == 1) & (model_1_preds == 0)).sum()

# f1-score
f1_model_1 = f1_score(y_train, model_1_preds)

# precision, recall, and support.
precision_model_1, recall_model_1, _, support_model_1 = precision_recall_fscore_support(
    y_train, model_1_preds)

In [16]:
print(f'accuracy for model 1: \t{accuracy_model_1}')
print(f'true positives for model 1: \t{tp_model_1}')
print(f'false positives for model 1: \t{fp_model_1}')
print(f'true negatives for model 1: \t{tn_model_1}')
print(f'false negatives for model 1: \t{fn_model_1}')
print(f'f1_score for model 1: \t{f1_model_1}')
print(f'precision for model 1: \t{precision_model_1}')
print(f'recall for model 1: \t{recall_model_1}')
print(f'support for model 1: \t{support_model_1}')

accuracy for model 1: 	0.9919678714859438
true positives for model 1: 	187
false positives for model 1: 	0
true negatives for model 1: 	307
false negatives for model 1: 	4
f1_score for model 1: 	0.9894179894179894
precision for model 1: 	[0.98713826 1.        ]
recall for model 1: 	[1.         0.97905759]
support for model 1: 	[307 191]


### what was actually being asked for here:

Accuracy: (TP + TN) / (TP + TN + FP + FN)

TPR: TP / (TP + FN)

FPR: FP / (FP + TN)

TNR: TN / (FP + TN)

FNR: FN / (FN + TP)

Precision: TP / (TP + FP)

Recall: TP / (TP + FN)

F1: 2 * (precision * recall) / (precision + recall)

support: sum of true positives, sum of true negatives

In [48]:
conf.ravel()

array([307,   0,   4, 187])

In [49]:
TN, FP, FN, TP = conf.ravel()

In [51]:
Accuracy= (TP + TN) / (TP + TN + FP + FN)

TPR = recall = TP / (TP + FN)

FPR= FP / (FP + TN)

TNR= TN / (FP + TN)

FNR= FN / (FN + TP)
    
precision= TP / (TP + FP)
    
F1= 2 * (precision * recall) / (precision + recall)

In [52]:
TPR

0.9790575916230366

In [53]:
print(f'accuracy for model 1: \t{accuracy_model_1}')
print(f'true positives for model 1: \t{TPR}')
print(f'false positives for model 1: \t{FPR}')
print(f'true negatives for model 1: \t{TNR}')
print(f'false negatives for model 1: \t{FNR}')
print(f'f1_score for model 1: \t{f1_model_1}')
print(f'precision for model 1: \t{precision_model_1}')
print(f'recall for model 1: \t{recall_model_1}')
print(f'support for model 1: \t{support_model_1}')

accuracy for model 1: 	0.9919678714859438
true positives for model 1: 	0.9790575916230366
false positives for model 1: 	0.0
true negatives for model 1: 	1.0
false negatives for model 1: 	0.020942408376963352
f1_score for model 1: 	0.9894179894179894
precision for model 1: 	[0.98713826 1.        ]
recall for model 1: 	[1.         0.97905759]
support for model 1: 	[307 191]


## 5.
Run through steps 2-4 using a different max_depth value.

In [17]:
# make the decision tree classifier object
clf2 = DecisionTreeClassifier(max_depth=4)
# fit the DTC object
clf2.fit(X_train, y_train)
# use the DTC object
model_2_preds = clf2.predict(X_train)

In [18]:
# model 2 score
print(f'model 2 accuracy is: {clf2.score(X_train, y_train):.2%}\n')

# confusion matrix for model 2
print(f'confusion matrix for model 2: \n{confusion_matrix(y_train, model_2_preds)}\n')

# classification report for model 2
print(f'classification report for model 2:\n\n \
{classification_report(y_train, model_2_preds)}')

model 2 accuracy is: 84.74%

confusion matrix for model 2: 
[[290  17]
 [ 59 132]]

classification report for model 2:

               precision    recall  f1-score   support

           0       0.83      0.94      0.88       307
           1       0.89      0.69      0.78       191

    accuracy                           0.85       498
   macro avg       0.86      0.82      0.83       498
weighted avg       0.85      0.85      0.84       498



In [19]:
# Accuracy
accuracy_model_2 = accuracy_score(y_train, model_2_preds)

# true positive rate
tp_model_2 = ((y_train == 1) & (model_2_preds == 1)).sum()

# false positive rate
fp_model_2 = ((y_train == 0) & (model_2_preds == 1)).sum()

# true negative rate
tn_model_2 = ((y_train == 0) & (model_2_preds == 0)).sum()

# false negative rate
fn_model_2 = ((y_train == 1) & (model_2_preds == 0)).sum()

# f1-score
f1_model_2 = f1_score(y_train, model_2_preds)

# precision, recall, and support.
precision_model_2, recall_model_2, _, support_model_2 = precision_recall_fscore_support(
    y_train, model_2_preds)

In [20]:
print(f'accuracy for model 2: \t{accuracy_model_2}')
print(f'true positives for model 2: \t{tp_model_2}')
print(f'false positives for model 2: \t{fp_model_2}')
print(f'true negatives for model 2: \t{tn_model_2}')
print(f'false negatives for model 2: \t{fn_model_2}')
print(f'f1_score for model 2: \t{f1_model_2}')
print(f'precision for model 2: \t{precision_model_2}')
print(f'recall for model 2: \t{recall_model_2}')
print(f'support for model 2: \t{support_model_2}')

accuracy for model 2: 	0.8473895582329317
true positives for model 2: 	132
false positives for model 2: 	17
true negatives for model 2: 	290
false negatives for model 2: 	59
f1_score for model 2: 	0.7764705882352941
precision for model 2: 	[0.83094556 0.88590604]
recall for model 2: 	[0.94462541 0.69109948]
support for model 2: 	[307 191]


In [21]:
# make the decision tree classifier object
clf3 = DecisionTreeClassifier(max_depth=5)
# fit the DTC object
clf3.fit(X_train, y_train)
# use the DTC object
model_3_preds = clf3.predict(X_train)

In [22]:
# model 3 score
print(f'model 3 accuracy is: {clf3.score(X_train, y_train):.2%}\n')

# confusion matrix for model 3
print(f'confusion matrix for model 3: \n{confusion_matrix(y_train, model_3_preds)}\n')

# classification report for model 3
print(f'classification report for model 3:\n\n \
{classification_report(y_train, model_3_preds)}')

model 3 accuracy is: 85.34%

confusion matrix for model 3: 
[[292  15]
 [ 58 133]]

classification report for model 3:

               precision    recall  f1-score   support

           0       0.83      0.95      0.89       307
           1       0.90      0.70      0.78       191

    accuracy                           0.85       498
   macro avg       0.87      0.82      0.84       498
weighted avg       0.86      0.85      0.85       498



In [23]:
# Accuracy
accuracy_model_3 = accuracy_score(y_train, model_3_preds)

# true positive rate
tp_model_3 = ((y_train == 1) & (model_3_preds == 1)).sum()

# false positive rate
fp_model_3 = ((y_train == 0) & (model_3_preds == 1)).sum()

# true negative rate
tn_model_3 = ((y_train == 0) & (model_3_preds == 0)).sum()

# false negative rate
fn_model_3 = ((y_train == 1) & (model_3_preds == 0)).sum()

# f1-score
f1_model_3 = f1_score(y_train, model_3_preds)

# precision, recall, and support.
precision_model_3, recall_model_3, _, support_model_3 = precision_recall_fscore_support(
    y_train, model_3_preds)

In [24]:
print(f'accuracy for model 3: \t{accuracy_model_3}')
print(f'true positives for model 3: \t{tp_model_3}')
print(f'false positives for model 3: \t{fp_model_3}')
print(f'true negatives for model 3: \t{tn_model_3}')
print(f'false negatives for model 3: \t{fn_model_3}')
print(f'f1_score for model 3: \t{f1_model_3}')
print(f'precision for model 3: \t{precision_model_3}')
print(f'recall for model 3: \t{recall_model_3}')
print(f'support for model 3: \t{support_model_3}')

accuracy for model 3: 	0.8534136546184738
true positives for model 3: 	133
false positives for model 3: 	15
true negatives for model 3: 	292
false negatives for model 3: 	58
f1_score for model 3: 	0.784660766961652
precision for model 3: 	[0.83428571 0.89864865]
recall for model 3: 	[0.95114007 0.69633508]
support for model 3: 	[307 191]


# bonus
lets make a bunch of models all at once and then display the scores all at once to find the best ALL AT ONCE.

In [57]:
model_dict = {}
for i in range(2,15):
    clf = DecisionTreeClassifier(max_depth=i)
    clf.fit(X_train, y_train)
    model_dict[f'dt_depth_{i}'] = {}
    model_dict[f'dt_depth_{i}']['model'] = clf
    model_dict[f'dt_depth_{i}']['train_score'] = clf.score(X_train, y_train)
    model_dict[f'dt_depth_{i}']['val_score'] = clf.score(X_val, y_val)
    model_dict[f'dt_depth_{i}']['val_diff'] = \
    clf.score(X_train, y_train) - clf.score(X_val, y_val)

In [58]:
model_dict.keys()

dict_keys(['dt_depth_2', 'dt_depth_3', 'dt_depth_4', 'dt_depth_5', 'dt_depth_6', 'dt_depth_7', 'dt_depth_8', 'dt_depth_9', 'dt_depth_10', 'dt_depth_11', 'dt_depth_12', 'dt_depth_13', 'dt_depth_14'])

In [59]:
model_dict['dt_depth_2']

{'model': DecisionTreeClassifier(max_depth=2),
 'train_score': 0.7951807228915663,
 'val_score': 0.7710280373831776,
 'val_diff': 0.02415268550838867}

In [63]:
for model in model_dict:
    print(model)
    print(model_dict[model]['train_score'])
    print(model_dict[model]['val_diff'])

dt_depth_2
0.7951807228915663
0.02415268550838867
dt_depth_3
0.8373493975903614
0.04295687422587546
dt_depth_4
0.8473895582329317
0.03430544608339903
dt_depth_5
0.8534136546184738
0.04032954246894116
dt_depth_6
0.8775100401606426
0.059753030814848174
dt_depth_7
0.891566265060241
0.06913635851818489
dt_depth_8
0.9096385542168675
0.1012273392635964
dt_depth_9
0.929718875502008
0.12598055774499872
dt_depth_10
0.9417670682730924
0.1380287505160831
dt_depth_11
0.9558232931726908
0.17077656420072818
dt_depth_12
0.9698795180722891
0.20819727508163488
dt_depth_13
0.9799196787148594
0.2088916413316818
dt_depth_14
0.9879518072289156
0.1842134894719063


## 6.
Which model performs better on your in-sample data?

The first model appears to perform better on the training data, with a 99.2% accuracy. But I think that this will have been overfit, and will not generalize well.

The second best model appears to be model 3 with max_depth=5

## 7.
Which model performs best on your out-of-sample data, the validate set?

In [25]:
X_val = validate[X_cols]
y_val = validate[y_cols]

In [26]:
val_1_preds = clf1.predict(X_val)
val_2_preds = clf2.predict(X_val)
val_3_preds = clf3.predict(X_val)

In [27]:
# model 1 score to validation
print(f'model 1 accuracy to the validation data is: {clf1.score(X_val, y_val):.2%}\n')

# confusion matrix for model 1
print(f'confusion matrix for model 1 to the validation data: \
\n{confusion_matrix(y_val, val_1_preds)}\n')

# classification report for model 1
print(f'classification report for model 1 to the validation data:\n\n \
{classification_report(y_val, val_1_preds)}')

model 1 accuracy to the validation data is: 74.30%

confusion matrix for model 1 to the validation data: 
[[96 36]
 [19 63]]

classification report for model 1 to the validation data:

               precision    recall  f1-score   support

           0       0.83      0.73      0.78       132
           1       0.64      0.77      0.70        82

    accuracy                           0.74       214
   macro avg       0.74      0.75      0.74       214
weighted avg       0.76      0.74      0.75       214



In [28]:
# model 2 score to validation
print(f'model 2 accuracy to the validation data is: {clf2.score(X_val, y_val):.2%}\n')

# confusion matrix for model 2
print(f'confusion matrix for model 2 to the validation data: \
\n{confusion_matrix(y_val, val_2_preds)}\n')

# classification report for model 2
print(f'classification report for model 2 to the validation data:\n\n \
{classification_report(y_val, val_2_preds)}')

model 2 accuracy to the validation data is: 81.31%

confusion matrix for model 2 to the validation data: 
[[119  13]
 [ 27  55]]

classification report for model 2 to the validation data:

               precision    recall  f1-score   support

           0       0.82      0.90      0.86       132
           1       0.81      0.67      0.73        82

    accuracy                           0.81       214
   macro avg       0.81      0.79      0.79       214
weighted avg       0.81      0.81      0.81       214



In [29]:
# model 3 score to validation
print(f'model 3 accuracy to the validation data is: {clf3.score(X_val, y_val):.2%}\n')

# confusion matrix for model 3
print(f'confusion matrix for model 3 to the validation data: \
\n{confusion_matrix(y_val, val_3_preds)}\n')

# classification report for model 3
print(f'classification report for model 3 to the validation data:\n\n \
{classification_report(y_val, val_3_preds)}')

model 3 accuracy to the validation data is: 81.31%

confusion matrix for model 3 to the validation data: 
[[119  13]
 [ 27  55]]

classification report for model 3 to the validation data:

               precision    recall  f1-score   support

           0       0.82      0.90      0.86       132
           1       0.81      0.67      0.73        82

    accuracy                           0.81       214
   macro avg       0.81      0.79      0.79       214
weighted avg       0.81      0.81      0.81       214



Models 2 (max_depth=4) and Model 3 (max_depth=5) appear to have performed the same on the validation data. Model 1 had the largest drop in performance, most likely due to overfitting.

Let's try out the test data on Model 2 since it had the lowest change in validation accuracy to the train data accuracy

In [30]:
X_test = test[X_cols]
y_test = test[y_cols]

In [31]:
test_2_preds = clf2.predict(X_test)

In [44]:
# model 2 score to test
print(f'model 2 accuracy to the test data is: {clf2.score(X_test, y_test):.2%}\n')

# confusion matrix for model 2
print(f'confusion matrix for model 2 to the test data: \
\n{confusion_matrix(y_test, test_2_preds)}\n')

# classification report for model 2
print(f'classification report for model 2 to the test data:\n\n \
{classification_report(y_test, test_2_preds)}')

model 2 accuracy to the test data is: 83.80%

confusion matrix for model 2 to the test data: 
[[99 11]
 [18 51]]

classification report for model 2 to the test data:

               precision    recall  f1-score   support

           0       0.85      0.90      0.87       110
           1       0.82      0.74      0.78        69

    accuracy                           0.84       179
   macro avg       0.83      0.82      0.83       179
weighted avg       0.84      0.84      0.84       179



Model 2 performed about the same on the test data as it did the training and validation data. Yay

# Telco data

## 1. 
Work through these same exercises using the Telco dataset.

In [32]:
train_telco, validate_telco, test_telco = prepare.wrangle_data('telco')

In [33]:
train_telco.head()

gender  senior_citizen  partner  dependents  tenure  phone_service  \
1843    Male               0        0           0       1              1   
5717  Female               0        1           1      62              1   
1083    Male               1        1           0      70              1   
5210  Female               0        0           1      31              1   
2475  Female               0        1           1      55              1   

     multiple_lines      online_security        online_backup  \
1843             No  No internet service  No internet service   
5717             No                  Yes                   No   
1083             No                  Yes                  Yes   
5210             No  No internet service  No internet service   
2475            Yes                  Yes                  Yes   

        device_protection  ... streaming_tv_yes  \
1843  No internet service  ...                0   
5717                  Yes  ...                0   
1083                  Yes  ...                1   
5210  No internet service  ...                0   
2475                   No  ...                0   

     streaming_movies_no_internet_service streaming_movies_yes  \
1843                                    1                    0   
5717                                    0                    0   
1083                                    0                    1   
5210                                    1                    0   
2475                                    0                    0   

      contract_type_one_year  contract_type_two_year  \
1843                       0                       0   
5717                       1                       0   
1083                       0                       1   
5210                       1                       0   
2475                       0                       0   

      internet_service_type_fiber_optic  internet_service_type_none  \
1843                                  0                           1   
5717                                  0                           0   
1083                                  1                           0   
5210                                  0                           1   
2475                                  1                           0   

     payment_type_credit_card_automatic payment_type_electronic_check  \
1843                                  0                             0   
5717                                  1                             0   
1083                                  1                             0   
5210                                  1                             0   
2475                                  0                             1   

     payment_type_mailed_check  
1843                         1  
5717                         0  
1083                         0  
5210                         0  
2475                         0  

[5 rows x 42 columns]

In [34]:
train_telco.churn.value_counts()

0    2897
1    1046
Name: churn, dtype: int64

In [35]:
# baseline will be churn = 0

train_telco['baseline'] = 0
validate_telco['baseline'] = 0
test_telco['baseline'] = 0

In [36]:
# What is your baseline accuracy?
print(f'baseline prediction accuracy score is: \
{accuracy_score(train_telco.churn, train_telco.baseline):.2%}')

baseline prediction accuracy score is: 73.47%


## 2. 
Fit the decision tree classifier to your training sample and transform (i.e. make predictions on the training sample)

In [37]:
train_telco.dtypes == 'O'

gender                                    True
senior_citizen                           False
partner                                  False
dependents                               False
tenure                                   False
phone_service                            False
multiple_lines                            True
online_security                           True
online_backup                             True
device_protection                         True
tech_support                              True
streaming_tv                              True
streaming_movies                          True
paperless_billing                        False
monthly_charges                          False
total_charges                            False
churn                                    False
contract_type                             True
internet_service_type                     True
payment_type                              True
gender_male                              False
multiple_line

In [38]:
X_cols = train_telco.drop(columns=[
     'gender',
     'multiple_lines',
     'online_security',
     'online_backup',
     'device_protection',
     'tech_support',
     'streaming_tv',
     'streaming_movies',
     'contract_type',
     'internet_service_type',
     'payment_type'
     ]).columns.to_list()
y_cols = 'churn'

In [39]:
X_train_telco = train_telco[X_cols]
y_train_telco = train_telco[y_cols]

In [40]:
# make the decision tree classifier object
clf1_telco = DecisionTreeClassifier()
# fit the DTC object
clf1_telco.fit(X_train_telco, y_train_telco)
# use the DTC object
model_1_telco_preds = clf1_telco.predict(X_train_telco)

## 3.
Evaluate your in-sample results using the model score, confusion matrix, and classification report.

In [41]:
# model 1 score
print(f'model 1 accuracy is: {clf1_telco.score(X_train_telco, y_train_telco):.2%}\n')

# confusion matrix for model 1
print(f'confusion matrix for model 1: \
\n{confusion_matrix(y_train_telco, model_1_telco_preds)}\n')

# classification report for model 1
print(f'classification report for model 1:\n\n \
{classification_report(y_train_telco, model_1_telco_preds)}')

model 1 accuracy is: 100.00%

confusion matrix for model 1: 
[[2897    0]
 [   0 1046]]

classification report for model 1:

               precision    recall  f1-score   support

           0       1.00      1.00      1.00      2897
           1       1.00      1.00      1.00      1046

    accuracy                           1.00      3943
   macro avg       1.00      1.00      1.00      3943
weighted avg       1.00      1.00      1.00      3943



In [42]:
# Accuracy
accuracy_model_1_telco = accuracy_score(y_train_telco, model_1_telco_preds)

# true positive rate
tp_model_1_telco = ((y_train_telco == 1) & (model_1_telco_preds == 1)).sum()

# false positive rate
fp_model_1_telco = ((y_train_telco == 0) & (model_1_telco_preds == 1)).sum()

# true negative rate
tn_model_1_telco = ((y_train_telco == 0) & (model_1_telco_preds == 0)).sum()

# false negative rate
fn_model_1_telco = ((y_train_telco == 1) & (model_1_telco_preds == 0)).sum()

# f1-score
f1_model_1_telco = f1_score(y_train_telco, model_1_telco_preds)

# precision, recall, and support.
precision_model_1_telco, recall_model_1_telco, _, support_model_1_telco = precision_recall_fscore_support(
    y_train_telco, model_1_telco_preds)

In [43]:
print(f'accuracy for model 1_telco: \t{accuracy_model_1_telco}')
print(f'true positives for model 1_telco: \t{tp_model_1_telco}')
print(f'false positives for model 1_telco: \t{fp_model_1_telco}')
print(f'true negatives for model 1_telco: \t{tn_model_1_telco}')
print(f'false negatives for model 1_telco: \t{fn_model_1_telco}')
print(f'f1_score for model 1_telco: \t{f1_model_1_telco}')
print(f'precision for model 1_telco: \t{precision_model_1_telco}')
print(f'recall for model 1_telco: \t{recall_model_1_telco}')
print(f'support for model 1_telco: \t{support_model_1_telco}')

accuracy for model 1_telco: 	1.0
true positives for model 1_telco: 	1046
false positives for model 1_telco: 	0
true negatives for model 1_telco: 	2897
false negatives for model 1_telco: 	0
f1_score for model 1_telco: 	1.0
precision for model 1_telco: 	[1. 1.]
recall for model 1_telco: 	[1. 1.]
support for model 1_telco: 	[2897 1046]
